python/cfl.py

```diff
@@ -6,36 +6,69 @@
 # 2013 Martin Uecker <uecker@eecs.berkeley.edu>
 # 2015 Jonathan Tamir <jtamir@eecs.berkeley.edu>
 
+import os  # Miscellaneous operating system interfaces
+import numpy as np  # NumPy (multidimensional numerical arrays library)
 
-import numpy as np
-
-def readcfl(name):
-    # get dims from .hdr
-    h = open(name + ".hdr", "r")
-    h.readline() # skip
-    l = h.readline()
-    h.close()
-    dims = [int(i) for i in l.split( )]
-
-    # remove singleton dimensions from the end
-    n = np.prod(dims)
-    dims_prod = np.cumprod(dims)
-    dims = dims[:np.searchsorted(dims_prod, n)+1]
-
-    # load data and reshape into dims
-    d = open(name + ".cfl", "r")
-    a = np.fromfile(d, dtype=np.complex64, count=n);
-    d.close()
-    return a.reshape(dims, order='F') # column-major
-
-	
-def writecfl(name, array):
-    h = open(name + ".hdr", "w")
-    h.write('# Dimensions\n')
-    for i in (array.shape):
-            h.write("%d " % i)
-    h.write('\n')
-    h.close()
-    d = open(name + ".cfl", "w")
-    array.T.astype(np.complex64).tofile(d) # tranpose for column-major order
-    d.close()
+# ======================================================================
+def readcfl(
+        basename,
+        dirpath='.'):
+    """
+    Read CFL header+data pair
+
+    Args:
+        basename (str): the base filename. Corresponding '.hdr' and '.cfl' files
+            must exist
+        dirpath (str): the working directory
+
+    Returns:
+        array (ndarray): The CFL data
+    """
+    filepath = os.path.join(dirpath, basename)
+
+    # load header
+    with open(filepath + '.hdr', 'r') as header_file:
+        header_file.readline()  # skip comment line
+        dim_line = header_file.readline()
+
+    # obtain the shape of the image
+    shape = [int(i) for i in dim_line.split(' ') if int(i) > 1]
```

Comment:
This also seems like a pretty fundamental change: If I understand it correctly, it squeezes the shape. That does not seem appropriate, since dimension position has meaning in bart. The old readcfl only squeezed away final singleton dimensions.